File: examples/modulith-clinical-tool-jpa/src/main/java/io/zenwave360/example/clinicaltool/modules/clinical/core/inbound/PatientsService.java
package io.zenwave360.example.clinicaltool.modules.clinical.core.inbound;

import io.zenwave360.example.clinicaltool.modules.clinical.core.domain.*;
import io.zenwave360.example.clinicaltool.modules.clinical.core.inbound.dtos.*;
import java.math.*;
import java.time.*;
import java.util.*;

/**
 * Inbound Service Port for managing [Patient, ProvisionalPatient].
 */
@org.springframework.modulith.NamedInterface("PatientsService")
public interface PatientsService {

    /**
     *
     *
     */
    public Patient loadPatient(String phoneNumber, String hisNumber);
    /**
     *
     *
     * With Events: [PatientCreated].
     *
     */
    public Patient createPatient(Patient input);
    /**
     *
     *
     */
    public Patient partialPatientUpdate(String phoneNumber, String hisNumber, java.util.Map input);
    /**
     *
     *
     */
    public Optional<Patient> updatePatient(Long id, Patient input);
    /**
     *
     *
     */
    public Optional<Patient> getPatient(Long id);
    /**
     *
     *
     */
    public PatientProfile getPatientProfileById(Long id);
    /**
     *
     *
     */
    public void associateDocumentWithPatient(DocumentSignatureRequestedInput input);
}

File: examples/modulith-klinical-tool-jpa/src/main/kotlin/io/zenwave360/example/clinicaltool/modules/clinical/core/inbound/PatientsService.kt
package io.zenwave360.example.clinicaltool.modules.clinical.core.inbound

import io.zenwave360.example.clinicaltool.modules.clinical.core.domain.*
import io.zenwave360.example.clinicaltool.modules.clinical.core.inbound.dtos.*
import java.math.*
import java.time.*

/** Inbound Service Port for managing [Patient]. */
@org.springframework.modulith.NamedInterface("PatientsService")
interface PatientsService {

    /**  */
    fun loadPatient(phoneNumber: String, hisNumber: String): Patient

    /** With Events: [PatientCreated]. */
    fun createPatient(input: Patient): Patient

    /**  */
    fun partialPatientUpdate(phoneNumber: String, hisNumber: String, input: Map<String, Any?>): Patient

    /**  */
    fun updatePatient(id: Long, input: Patient): Patient?

    /**  */
    fun getPatient(id: Long): Patient?

    /**  */
    fun getPatientProfileById(id: Long): PatientProfile

    /**  */
    fun associateDocumentWithPatient(input: DocumentSignatureRequestedInput): Unit
}
